core/src/main/java/org/elasticsearch/search/aggregations/InternalAggregation.java

```diff
@@ -140,6 +135,11 @@ public String getName() {
      */
     public abstract Type type();
 
+    @Override
+    public String getWriteableName() {
+        return type().stream().toUtf8();
```

Comment:
I get that , but it feels weird that we use it as a name to register things? You confirm that is correct?